src/stateManager/ValidationService.ts

```diff
@@ -1,384 +1,488 @@
+            this.storage.blocks.storeBlockConfirmation(blockConfirmation);
+            return ExecutionFlags.BROADCAST;
+        }
 
-        return success();
+        return null;
     }
 
-    public async validateDispute(
-        disputeStruct: DisputeStruct,
-        timestamp: Timestamp
-    ): Promise<boolean> {
-        // triggered by StateManager.onDisputeCommitted, which is triggered by chain event 'DisputeCommitted'
-        // therefore it is assumed that validation has already happened on
-
-        // this is the place to add any validation that should run on a new dispute
+    private checkConflictingBlock(
+        block: Block,
+        maybePreExistingSignedBlock: SignedBlockStruct | undefined
+    ): {
+        conflict: boolean;
+        fraudType?: FraudType;
+    } {
+        // conflicting block ?
+        if (!maybePreExistingSignedBlock) {
+            return { conflict: false };
+        }
 
-        // Note: disputeIndex property doesn't exist on DisputeStruct, removing this validation
-        // If dispute ordering validation is needed, it should be implemented differently
+        // name change for calirty, it isn't a maybe anymore
+        const conflictingSignedBlock = maybePreExistingSignedBlock;
 
-        const allowedParticipantsSet = await getActiveParticipants(
-            this.scmContract,
-            this.getChannelId()
+        const conflictingBlock = Block.decode(
+            conflictingSignedBlock.encodedBlock
         );
-        if (!allowedParticipantsSet.has(disputeStruct.disputer)) {
-            return false;
-        }
 
-        return true;
+        if (conflictingBlock.author === block.author) {
+            // DOUBLE SIGN
+            return {
+                conflict: true,
+                fraudType: FraudType.DOUBLE_SIGN
+            };
+        }
+        return {
+            conflict: true,
+            //If not linked we can't slash since the peer could have been building on the wrong 'reality' since someone performed a double sign
+            fraudType: this.isLinked(block)
+                ? FraudType.INVALID_STATE_TRANSITION
+                : undefined
+        };
     }
 
-    public async validateDisputeConfirmation(
-        disputeStruct: DisputeStruct,
-        confirmationSignature: Signature
-    ): Promise<ValidationResult> {
-        if (!this.isChannelOpen()) return notReady();
-
-        // Check if dispute exists
-        if (!this.agreementManager.isDisputeKnown(disputeStruct)) {
-            // Dispute not found - could be gossip arrived before event
-            // Return NOT_READY to allow retry
-            return notReady();
-        }
+    private async isDisputedFork(
+        forkId: ForkId,
+        channelId: ChannelId
+    ): Promise<boolean> {
+        return (
+            this.storage.disputes.getDisputedFork(forkId) ||
+            (await this.stateChannelManagerContract.isForkDisputed(
```

Comment:
this should be localEvm.diamond.isForkDisputed - we don't want to spam the RPC node and should do as much of the computation locaaly as possible